explorer/templates.go

```diff
@@ -303,6 +304,37 @@ func makeTemplateFuncMap(params *chaincfg.Params) template.FuncMap {
 		"toLowerCase": func(a string) string {
 			return strings.ToLower(a)
 		},
+		"fetchListNavURL": func(groupingStr string, interval int64) string {
+			// This allows all links in years page to be linked to months i.e.
+			// /year -> /months -> weeks -> /days -> /blocks
+			var matchedGrouping string
+			val := dbtypes.TimeGroupingFromStr(groupingStr)
+
+			switch val {
+			case dbtypes.YearGrouping:
+				matchedGrouping = "months"
+			case dbtypes.MonthGrouping:
+				matchedGrouping = "weeks"
+			case dbtypes.WeekGrouping:
+				matchedGrouping = "days"
+			// for dbtypes.DayGrouping and any other groupings default to blocks.
+			default:
+				return fmt.Sprintf("/blocks?offset=%d", interval)
+			}
+			matchingVal := dbtypes.TimeGroupingFromStr(matchedGrouping)
+			intervalVal, err := dbtypes.TimeBasedGroupingToInterval(matchingVal)
+			if err != nil {
+				log.Errorf("Resolving the new group interval failed: error : %v", err)
```

Comment:
Whenever the user has control over the input, and can cause a non-nil error at will, I prefer to have the log as Debug[f] or Trace[f]